src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/cartdiscounts/CartDiscountSyncIT.java

```diff
@@ -0,0 +1,161 @@
+package com.commercetools.sync.integration.ctpprojectsource.cartdiscounts;
+
+import com.commercetools.sync.cartdiscounts.CartDiscountSync;
+import com.commercetools.sync.cartdiscounts.CartDiscountSyncOptions;
+import com.commercetools.sync.cartdiscounts.CartDiscountSyncOptionsBuilder;
+import com.commercetools.sync.cartdiscounts.helpers.CartDiscountSyncStatistics;
+import io.sphere.sdk.cartdiscounts.AbsoluteCartDiscountValue;
+import io.sphere.sdk.cartdiscounts.CartDiscount;
+import io.sphere.sdk.cartdiscounts.CartDiscountDraft;
+import io.sphere.sdk.cartdiscounts.CartDiscountDraftBuilder;
+import io.sphere.sdk.cartdiscounts.CartPredicate;
+import io.sphere.sdk.cartdiscounts.ShippingCostTarget;
+import io.sphere.sdk.cartdiscounts.queries.CartDiscountQuery;
+import io.sphere.sdk.utils.MoneyImpl;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.commercetools.sync.commons.asserts.statistics.AssertionsForStatistics.assertThat;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.deleteCartDiscountsFromTargetAndSource;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.populateSourceProject;
+import static com.commercetools.sync.integration.commons.utils.CartDiscountITUtils.populateTargetProject;
+import static com.commercetools.sync.integration.commons.utils.ITUtils.deleteTypesFromTargetAndSource;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_SOURCE_CLIENT;
+import static com.commercetools.sync.integration.commons.utils.SphereClientUtils.CTP_TARGET_CLIENT;
+import static io.sphere.sdk.models.DefaultCurrencyUnits.EUR;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CartDiscountSyncIT {
+
+    /**
+     * Deletes types from the target CTP projects.
+     * Populates the target CTP project with test data.
+     */
+    @Before
+    public void setup() {
+        deleteTypesFromTargetAndSource();
+        deleteCartDiscountsFromTargetAndSource();
+        populateSourceProject();
+        populateTargetProject();
+    }
+
+    /**
+     * Deletes all the test data from the {@code CTP_SOURCE_CLIENT} and the {@code CTP_SOURCE_CLIENT} projects that
+     * were set up in this test class.
+     */
+    @AfterClass
+    public static void tearDown() {
+        deleteTypesFromTargetAndSource();
+        deleteCartDiscountsFromTargetAndSource();
+    }
+
+    @Test
+    public void sync_WithoutUpdates_ShouldReturnProperStatistics() {
+        // preparation
+        final List<CartDiscount> cartDiscounts = CTP_SOURCE_CLIENT
```

Comment:
Why do you need `CTP_SOURCE_CLIENT`? Can't you just create cartDiscounts directly without loading it from a project? Because in your test, you don't need `CTP_SOURCE_CLIENT` anyway.

Reply:
nice question, yes I can create the test scenario as you explain, but this way test explain some use case which is different than the externalsource package, the scenario that you explaining it could be in the externalsource package, here we are focused to sync between 2 projects, so first we are getting resources from source and synchronizing with the target. I hope, I understand what you meant in your question.

note: check the `populateSourceProject();` before all test, it's creating some resource both in source and target projects, in the source project, there are 2 cart discounts, cartDiscount1 and cartDiscount2, in the target it's cartDiscount1.